json-rpc/src/util.rs

```diff
@@ -98,6 +96,75 @@ impl SdkLang {
     }
 }
 
+pub static SDK_VERSION_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"\b([0-3])\.(\d{1,2})\.(\d{1,2})\b").unwrap());
+pub static DEFAULT_SDK_VERSION: Lazy<SdkVersion> = Lazy::new(|| SdkVersion {
+    major: 0,
+    minor: 0,
+    patch: 0,
+});
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct SdkVersion {
+    pub major: u16,
+    pub minor: u16,
+    pub patch: u16,
+}
+
+impl SdkVersion {
+    pub fn from_str(user_agent_part: &str) -> SdkVersion {
+        if let Some(captures) = SDK_VERSION_REGEX.captures(user_agent_part) {
+            if captures.len() == 4 {
+                // println!("{}   -   {}.{}.{}", captures[0].to_string(), captures[1].to_string(), captures[2].to_string(), captures[3].to_string());
+                if let Ok(major) = u16::from_str(&captures[1]) {
+                    if let Ok(minor) = u16::from_str(&captures[2]) {
+                        if let Ok(patch) = u16::from_str(&captures[3]) {
+                            return SdkVersion {
+                                major,
+                                minor,
+                                patch,
+                            };
+                        }
+                    }
+                }
+            }
+        }
+        *DEFAULT_SDK_VERSION
+    }
+
+    pub fn as_str(&self) -> String {
+        format!("{}.{}.{}", self.major, self.minor, self.patch)
+    }
+}
+
+pub static DEFAULT_SDK_INFO: Lazy<SdkInfo> = Lazy::new(|| SdkInfo {
+    language: SdkLang::Unknown,
+    version: *DEFAULT_SDK_VERSION,
+});
```

Comment:
Like above, implement the Default trait for SdkInfo instead of doing this probably by manually implementing it on SdkLang and then deriving it on SdkInfo